### PR TITLE
feat: check actual size after pin

### PR DESCRIPTION
**Pull request by nduchak**

### CODE REVIEW

**Comment by julianmrodri** [comment]
Should this close #124 ?

File: src/providers/ipfs.ts [hunk 1]
```diff
@@ -70,43 +72,97 @@ export class PinJob extends Job {
...
 
+export function getDagStat (nodeUrl: string): (cid: CID, options?: any) => Promise<any> {
+  return (cid: CID, options?: RequestInit): Promise<any> =>
+    fetch(`${nodeUrl}/dag/stat?arg=${cid.toString()}`, { method: 'POST', ...options })
+      .then(async res => {
+        if (!res.ok) {
+          throw new Error(`Get dag stat for hash ${cid.toString()} error, ${res.statusText}`)
+        }
+        return (await res.text())
+          .split('\n')
```
Comment (AuHau) [inline src/providers/ipfs.ts:159]: I am not convinced that this custom parsing is needed 😳
When I do:

```
$ curl -X POST "http://127.0.0.1:5001/api/v0/dag/stat?arg=QmUszxPhL5s7LHyn7ehbFTu9DmQUbbdKniTpJkQr7EpZgR&progress=true"
```

I get normal looking JSON:

```
{"Size":51,"NumBlocks":1}
```

What are you getting?

Comment (nduchak) [reply src/providers/ipfs.ts:159]: image
This what I am talking about

Comment (AuHau) [reply src/providers/ipfs.ts:159]: Hmm I believe that is because of the `progress=true` ;-) We don't need progress here, so just don't use the flag...

Comment (AuHau) [reply src/providers/ipfs.ts:159]: I see I had it in the CURL command there as well, only my data were small enough that it did not emit the progress... 

Try it without it, please :-)

Comment (nduchak) [reply src/providers/ipfs.ts:159]: Same result

Comment (nduchak) [reply src/providers/ipfs.ts:159]: image

Comment (AuHau) [reply src/providers/ipfs.ts:159]: Hmmm so it seems progress is true by default :-D So use `progress=false` ;-)

Comment (nduchak) [reply src/providers/ipfs.ts:159]: Ahh, yes. Thanks 👍

**Review by AuHau — APPROVED** [review]
LGTM 👍 Thanks!

**Pull request merged**